Add-IN/ACDC4Robot/core/link.py

```diff
@@ -69,18 +69,14 @@ def get_parent_joint(self) -> adsk.fusion.Joint:
         # TODO: in closed loop mechanism, there exits one assembly method that
         # make one link have two parent joint, write a instruction or fix this bug
         # text_palette = constants.get_text_palette()
-        joint_list: adsk.fusion.JointList = self.link.joints
```

Comment:
In some cases, when there are nested components, retrieving `link.joints` will throw the exception: "RuntimeError: 3 : object does not belong to the occurrence's component"

The solution is to fetch all the joints in the design and perform the same logic.